src/abstractdataframe/selection.jl

```diff
@@ -469,8 +479,52 @@ SELECT_ARG_RULES =
     It is not allowed to pass `renamecols=false` if `old_column` is empty
     as it would generate an empty column name.
 
-    Column renaming and transformation operations can be passed wrapped in
-    vectors (this is useful when combined with broadcasting).
+    # Rules when `new_column_name` is a vector of `Symbol` or a string or is `AsTable`
+
+    In this case it is assumed that `fun` returns multiple columns.
+
+    If `fun` returns one of `AbstractDataFrame`, `NamedTuple`, `DataFrameRow`,
+    `AbstractMatrix` then rules described below for `args` being a function or
+    a type apply.
+
+    If `fun` returns an `AbstractVector` then each element of this vector must
+    support `keys` function that must return a collection of `Symbol`s, strings
+    or integers; the return value of `keys` must be identical for all elements.
+    Then as many columns are created as there are elements in the return value
+    of the `keys` function and their names are set to be equal to the key names,
+    except if `keys` returns integers, in which case they are prefixed by `x`
+    (so the column names are e.g. `x1`, `x2`, ...)
+
+    If `fun` returns a value of any other type then it is assumed that it is
+    a table conforming to Tables.jl API and the `Tables.columntable` function is
+    called on it to get the resulting columns and their names.
+
+    Additionally if `new_column_name` is a vector of `Symbol` or string then column
+    names produced using the rules above are ignored and replaced by `new_column_name`
+    (the number of columns must be the same as the length `new_column_name` in this case).
+
+    # Rules when element of `args` is a function or a type
+
+    In this case a transformaton is passed `df` as a single argument.
+
+    If the return value of the transformation is of `AbstractDataFrame`,
+    `NamedTuple`, `DataFrameRow` or `AbstractMatrix` then it is treated as
+    containing multiple columns. For `AbstractMatrix` column names are generated
+    as `x1`, `x2`, etc. For `AbstractDataFrame`, `NamedTuple` of vectors and
+    `AbstractMatrix` the columns are taken as is from the returned value. For
+    `DataFrameRow` and` NamedTuple` not containing any vectors the returned
+    value is broadcasted a vector matching the target number of rows in the data
+    frame.
+
+    If the return value is an `AbstractVector` then it is used as-is. The resulting
+    column gets a name `x1`.
+
+    In all other cases the return value is broadcasted into a vector matching
+    the target number of rows in the data frame. As a particular rule, values
+    wrapped in a `Ref` or a `0`-dimensional `AbstractArray` are unwrapped and
```

Comment:
Is this a particular rule or just the application of broadcasting rules?

Reply:
So here maybe we need to change a wording if we want to be precise. I have written "broadcasted into" which suggests we use broadcasting from Base, but actually we do not. We rather "reuse the value", and with respect to this `Ref` and `0`-dimensional `AbstractArray` are special cases.

We do not do broadcasting, as it has a different set of corner cases (e.g. a 1-element vector would be expanded and we do not do it; if a returned value does not support broadcasting an error would be thrown and we reuse it etc.)

What do you think would be best here?

Reply:
In other words: we here use "pseduo-broadcasting" not "broadcasting" and it would be best to discuss how to document it in general (it is documented well I think for `DataFrame` and `insertcols!` where we also use "pseudo-broadcasting", but there additional rules apply (not all values are accepted), that we do not impose here)

Reply:
Maybe just say "repeated"?

Reply:
ok